pkg/netflow/common/flowtype.go

```diff
@@ -0,0 +1,71 @@
+package common
+
+import (
+	"fmt"
+)
+
+// FlowType represent the flow protocol (netflow5,netflow9,ipfix, sflow, etc)
+type FlowType string
+
+// Flow Types
+const (
+	TypeIPFIX    FlowType = "ipfix"
+	TypeSFlow5   FlowType = "sflow5"
+	TypeNetFlow5 FlowType = "netflow5"
+	TypeNetFlow9 FlowType = "netflow9"
+	TypeUnknown  FlowType = "unknown"
+)
+
+// FlowTypeDetails contain list of valid FlowTypeDetail
+var FlowTypeDetails = map[FlowType]FlowTypeDetail{
+	TypeIPFIX: {
+		name:        TypeIPFIX,
+		defaultPort: uint16(4739),
+	},
+	TypeSFlow5: {
+		name:        TypeSFlow5,
+		defaultPort: uint16(6343),
+	},
+	TypeNetFlow5: {
+		name:        TypeNetFlow5,
+		defaultPort: uint16(2055),
+	},
+	TypeNetFlow9: {
+		name:        TypeNetFlow9,
+		defaultPort: uint16(2055),
+	},
+}
+
+// FlowTypeDetail represent the flow protocol (netflow5,netflow9,ipfix, sflow, etc)
+type FlowTypeDetail struct {
+	name        FlowType
+	defaultPort uint16
+}
+
+// Name returns the flow type name
+func (f FlowTypeDetail) Name() FlowType {
+	return f.name
+}
+
+// DefaultPort returns the default port
+func (f FlowTypeDetail) DefaultPort() uint16 {
+	return f.defaultPort
+}
+
+// GetFlowTypeByName search FlowTypeDetail by name
+func GetFlowTypeByName(name FlowType) (FlowTypeDetail, error) {
+	detail, ok := FlowTypeDetails[name]
+	if !ok {
+		return FlowTypeDetail{}, fmt.Errorf("flow type `%s` is not valid", name)
+	}
+	return detail, nil
+}
+
+// GetAllFlowTypes returns all flow names
+func GetAllFlowTypes() []FlowType {
```

Comment:
Do we need both a public `FlowTypeDetails` and `GetAllFlowTypes` ? Should we make the first one private?

Reply:
> Should we make the first one private?

No sure to understand since it's currently used.

Do you see a better alternative ?

Reply:
Nevermind me, I read this fast and thought you were only providing a copy of `FlowTypeDetails` but you actually convert it to an array 👍